code/contracts/ONEWallet.sol

```diff
@@ -117,35 +118,66 @@ contract ONEWallet is TokenTracker {
         return nonces[index];
     }
 
-    function getCommits() external view returns (bytes32[] memory, bytes32[] memory, uint32[] memory, bool[] memory)
+    function getCommits() external pure returns (bytes32[] memory, bytes32[] memory, uint32[] memory, bool[] memory){
+        revert("Deprecated");
+    }
+
+    function getAllCommits() external view returns (bytes32[] memory, bytes32[] memory, bytes32[] memory, uint32[] memory, bool[] memory)
     {
-        bytes32[] memory paramHashes = new bytes32[](commits.length);
-        bytes32[] memory hashes = new bytes32[](commits.length);
-        uint32[] memory timestamps = new uint32[](commits.length);
-        bool[] memory completed = new bool[](commits.length);
+        uint32 numCommits = 0;
         for (uint32 i = 0; i < commits.length; i++) {
-            Commit storage c = commitLocker[commits[i]];
+            Commit[] storage cc = commitLocker[commits[i]];
+            numCommits += uint32(cc.length);
+        }
+        bytes32[] memory hashes = new bytes32[](numCommits);
+        bytes32[] memory paramHashes = new bytes32[](numCommits);
+        bytes32[] memory verificationHashes = new bytes32[](numCommits);
+        uint32[] memory timestamps = new uint32[](numCommits);
+        bool[] memory completed = new bool[](numCommits);
+        uint32 index = 0;
+        for (uint32 i = 0; i < commits.length; i++) {
+            Commit[] storage cc = commitLocker[commits[i]];
+            for (uint32 j = 0; j < cc.length; j++) {
+                Commit storage c = cc[j];
+                hashes[index] = c.hash;
```

Comment:
Hash can be removed from the Commit structure. It is never utilized for any verification, so storage can be saved. Moreover, the hash is a key to commitLocker map, so you already have that info.

Reply:
Discussed over chat and I agree.